src/Event/DoctrineDatasetListener.php

```diff
@@ -23,14 +24,23 @@ class DoctrineDatasetListener
      */
     protected $messageBus;
 
+    /**
+     * Memcached adapter instance.
+     *
+     * @var \Memcached
+     */
+    protected $cache;
+
     /**
      * DoctrineDatasetListener constructor.
      *
      * @param MessageBusInterface $messageBus Symfony messenger bus interface.
+     * @param MemcachedAdapter    $cache      Memcached adapter instance.
```

Comment:
Not a parameter. And because you calling a static method, you can construct this in the calling function.